HeterogeneousCore/CUDAUtilities/interface/maxCoopBlocks.h

```diff
@@ -0,0 +1,30 @@
+
+#ifndef HeterogeneousCore_CUDAUtilities_interface_maxCoopBlocks_h
+#define HeterogeneousCore_CUDAUtilities_interface_maxCoopBlocks_h
+
+#include <cuda_runtime.h>
+#include <algorithm>
+#include <cstdlib>
+
+template <typename F>
+inline int maxCoopBlocks(F kernel, int nthreads, int shmem, int device, int redFact = 10) {
```

Comment:
I checked: major slowdown

Reply:
This would go well with a more general development I've been thinking about for a while (and that Abdulla may work on, if he comes to CERN in January): making the launch configuration of each kernel configurable, with a common interface.